crates/sui-core/src/consensus_handler.rs

```diff
@@ -2009,6 +2044,87 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                     // TODO(fastpath): Handle unlocking.
                     continue;
                 }
+                // When pre-consensus locking is disabled, perform post-consensus owned object
+                // conflict detection BEFORE setting Finalized status. If lock acquisition fails,
+                // the transaction has invalid/conflicting owned inputs and should be dropped.
+                //
+                // During crash recovery, transactions may have already been executed (either via
+                // checkpoint sync or locally before the crash). Skip lock acquisition for already
+                // executed transactions to avoid incorrectly dropping them due to stale object
+                // versions (which were advanced by prior execution).
+                //
+                // Additionally, check if the transaction was already assigned to a checkpoint
+                // (in builder_digest_to_checkpoint). This handles the case where checkpoint
+                // summary was persisted but transaction effects weren't - we must not drop
+                // these transactions or we'll get checkpoint mismatch errors.
```

Comment:
I think this whole section is unnecessary (and most likely does not work correctly).

This is where you would read from the RefCountedHashmap (before also checking the db) to see if an earlier commit locked the objects. If you do that crash recovery should be seamless